lib/entities/guess.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+class Guess < ValidatedEntity
+  attr_reader :guess_code, :errors
+
+  SECRET_CODE_LENGTH = 4
+  ELEMENT_MIN_VALUE = 1
+  ELEMENT_MAX_VALUE = 6
+
+  HINT = 'hint'
+  PLUS = '+'
+  MINUS = '-'
+
+  def initialize(guess_code)
+    super()
+    @guess_code = guess_code
+  end
+
+  def validate
+    return if hint?
+
+    @errors << fault.secret_code_format unless check_integer
+    @errors << fault.secret_code_length(SECRET_CODE_LENGTH) unless check_length
+    @errors << fault.secret_code_digits_range(ELEMENT_MIN_VALUE, ELEMENT_MAX_VALUE) unless check_digits_range
+  end
+
+  def mark_guess(secret_code)
+    @secret_code = secret_code
+    cloned_code = @secret_code.clone
+    input_digits = convert_input(@guess_code)
+    marked_guess = []
```

Comment:
why do you need that variable?

Reply:
to fill it with '+' and '-', after convert it to string which i use like a guess result